scenarios/aws/microVMs/microvms/domain.go

```diff
@@ -38,8 +39,9 @@ type Domain struct {
 	lvDomain    *libvirt.Domain
 }
 
-func generateDomainIdentifier(vcpu, memory int, vmsetName, tag, arch string) string {
-	return fmt.Sprintf("ddvm-%s-%s-%s-%d-%d", vmsetName, arch, tag, vcpu, memory)
+func generateDomainIdentifier(vcpu, memory int, vmsetTags, tag, arch string) string {
+	setTagsWithDelim := fmt.Sprintf("%s-ddvm", vmsetTags)
+	return fmt.Sprintf("%s-%s-%s-%d-%d", arch, tag, setTagsWithDelim, vcpu, memory)
```

Comment:
what is the usage of `domainID`? is it used for filenames and potentially can exceed the `MAX_PATH` (255 chars)?

Reply:
`domainID` is used for naming the VMs and generating their resource URN. The URN has the 255 character limit and yes it can potentially exceed this limit. 
The problem is that we overload the VM name in the datadog-agent CI for matching a VM name as reported by libvirt to its corresponding VMset. This is why we have to incorporate the tag information in the `domainID`. The rest of the information, such as the `arch` and `tag` are required for generating a unique URN

Reply:
if we end up generating a longer than 255 string in `generateDomainIdentifier` we end up with an error later in the callstack?

should we consider adding "truncation" logic in this function to generate only valid names? (maybe performing some transformation on `setTagsWithDelim` as I assume this can be the main cause for exceeding the limit)

WDYT?

Reply:
The problem is that at this stage we do not know if we will be above the 255 character limit. The reason for this is that the underlying `test-infra-definitions` framework also adds strings to the resource URNs.

Secondly, truncation cannot work even if it was possible. The `domainID` is generated using user provided `tags`. Atleast in the datadog-agent CI we expect to find these tags in the domain name.